### DISCUSSION TITLE
Segmentation issues similar to #730

### DISCUSSION THREAD

**Discussion by mdifab** [neuropoly/axondeepseg#736] [Help needed]
Hi, 

I'm very new to ADS so I'm trying to segment some TEM images I have using Napari. I've read that your TEM model is optimised for 10 nm px size, while the pixel size I normally have in my images is 2-3 nm, so slightly smaller.

So far I tried different approaches but none seems to work for me, but I cannot figure out what's missing. Hereafter I list some of the things I tried at different steps.

- I used the original images I have specifying the px size in microns
- I resampled the original images to get px size of 10 nm that could match the model
- I changed the zoom factor to 0.5, 2, 3, 4

So far I end up either with a totally empty (black) myelin/axon segmentation or with some myelin/axon segmentation that does not correspond to the actual myelin/axon pattern in my image.

![Screenshot 2023-04-21 at 15 35 06](https://user-images.githubusercontent.com/119606403/233649476-51effe30-8d50-4170-a90a-da01315a6c6f.png)

In this case, the image I'm showing in the screenshot has a px size of 0.0024 micron and I was using the following settings
![Screenshot 2023-04-21 at 15 37 06](https://user-images.githubusercontent.com/119606403/233649838-e4ac8d5c-8d4d-404f-9d53-fb78cbd84573.png)

but I had similar results when I changed the zoom factor or when I resampled the image to get a px size of 10 nm.

PS the images I'm using are not single field of view images but they're all tilings of single images with smaller fields of view. I guess it's the same for the training images and it's not a problem, right?

Could you please help me with that?

Thanks a lot

**Comment by mathieuboudreau** [comment]
@mdifab Could you send us one of your original (unresampled) images to axondeepseg@googlegroups.com along with the resolution for that image, so that one of us can try and see if there's a setting that would work better for you?

**Reply by mdifab** [reply]
Sure, just done it. Thanks!

**Reply by mathieuboudreau** [reply]
Thanks for that @mdifab !

I was able to reproduce your problem locally with your images.

With the help of your images, I discovered a bug in our software related to the data precision (i.e. bit depth) of input images. Your images have 16-bit integer precision (i.e. values of grey can go from 0 to ~65000). It seems our conversion to 8 bit (needed to segment using our model) is not working properly. If I manually convert your images to 8 bit (using an image tool like Gimp), then I get results that are much closer to expected:
<img width="1417" alt="Screenshot 2023-04-21 at 1 50 25 PM" src="https://user-images.githubusercontent.com/1421029/233692191-f4fb78b8-a991-4ce6-b32e-690d20837a7f.png">

In the short term, a solution for you would be to convert your images manually to 8 bits prior to running our software with them. eg https://www.youtube.com/watch?v=exLmtLBrgdY

I'll open an issue to fix the bug, and will notify you here when the problem is resolved.
